store/src/commonMain/kotlin/org/mobilenativefoundation/store/store5/impl/RealStore.kt

```diff
@@ -89,6 +89,19 @@ internal class RealStore<Key : Any, Network : Any, Output : Any, Local : Any>(
                 // if we read a value from cache, dispatch it first
                 emit(StoreReadResponse.Data(value = it, origin = StoreReadResponseOrigin.Cache))
             }
+
+            if (sourceOfTruth == null && !request.fetch) {
+                if (memCache == null) {
+                    emit(StoreReadResponse.Error.Exception(
+                        error = IllegalStateException("Cache-only request made with no cache configured"),
+                        origin = StoreReadResponseOrigin.Cache
+                    ))
```

Comment:
Because this is a developer error and not a data loading error, I think we should log and return `NoNewData`